From 7a945d31aeb34fc73cf86a494673ae97e069d84d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim=20R=C3=BChsen?= <tim.ruehsen@gmx.de>
Date: Sun, 30 Jun 2024 19:33:01 +0200
Subject: [PATCH] Disable TCP Fast Open by default

* docs/wget2.md: Amended description of --tcp-fastopen.
* src/options.c (struct config config): Disabled TFO.
---
 docs/wget2.md | 8 +++++++-
 src/options.c | 1 -
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/docs/wget2.md b/docs/wget2.md
index 61da3ccb..06828bf8 100644
--- a/docs/wget2.md
+++ b/docs/wget2.md
@@ -730,12 +730,18 @@ Go to background immediately after startup. If no output file is specified via t
 
 ### `--tcp-fastopen`
 
-  Enable support for TCP Fast Open (TFO) (default: on).
+  Enable support for TCP Fast Open (TFO) (default: off).
 
   TFO reduces connection latency by 1 RT on "hot" connections (2nd+ connection to the same host in a certain amount of time).
 
   Currently this works on recent Linux and OSX kernels, on HTTP and HTTPS.
 
+  The main reasons why TFO is disabled by default are
+    - possible user tracking issues
+    - possible issues with middle boxes that do not support TFO
+
+  This article gives has more details about TFO than fits here: https://candrews.integralblue.com/2019/03/the-sad-story-of-tcp-fast-open/
+
 ### `--dns-cache-preload=file`
 
   Load a list of IP / Name tuples into the DNS cache.
diff --git a/src/options.c b/src/options.c
index 026aa415..f4b5d1a1 100644
--- a/src/options.c
+++ b/src/options.c
@@ -1235,7 +1235,6 @@ struct config config = {
 	.max_redirect = 20,
 	.max_threads = 5,
 	.dns_caching = 1,
-	.tcp_fastopen = 1,
 	.user_agent = PACKAGE_NAME"/"PACKAGE_VERSION,
 	.verbose = 1,
 	.check_certificate= CHECK_CERTIFICATE_ENABLED,
-- 
2.43.0

